src/common/low_precision_transformations/tests/gather_transformation.cpp

```diff
@@ -0,0 +1,219 @@
+
+    class Expected {
+    public:
+        ngraph::element::Type precisionBeforeDequantization;
+        ngraph::builder::subgraph::DequantizationOperations dequantizationBefore;
+        ngraph::element::Type precisionAfterOperation;
+        ngraph::builder::subgraph::DequantizationOperations dequantizationAfter;
+    };
+
+    std::vector<int> gatherIndicesValues;
+    std::vector<int> axis;
+    TestTransformationParams params;
+    Actual actual;
+    Expected expected;
+};
+
+typedef std::tuple<
+    ngraph::PartialShape,
+    GatherTransformationTestValues
+> GatherTransformationParams;
+
+class GatherTransformation : public LayerTransformation, public testing::WithParamInterface<GatherTransformationParams> {
+public:
+    void SetUp() override {
+        const ngraph::PartialShape inputShape = std::get<0>(GetParam());
+        const GatherTransformationTestValues testValues = std::get<1>(GetParam());
+
+        actualFunction = ngraph::builder::subgraph::GatherFunction::getOriginal(
+            inputShape,
+            testValues.gatherIndicesValues,
+            testValues.axis,
+            testValues.actual.precisionBeforeDequantization,
+            testValues.actual.dequantization);
+
+        SimpleLowPrecisionTransformer transformer;
+        transformer.add<ngraph::pass::low_precision::GatherTransformation, ngraph::opset1::Gather>(testValues.params);
+        transformer.transform(actualFunction);
+
+        referenceFunction = ngraph::builder::subgraph::GatherFunction::getReference(
+            inputShape,
+            testValues.gatherIndicesValues,
+            testValues.axis,
+            testValues.expected.precisionBeforeDequantization,
+            testValues.expected.dequantizationBefore,
+            testValues.expected.precisionAfterOperation,
+            testValues.expected.dequantizationAfter);
+    }
+
+    static std::string getTestCaseName(testing::TestParamInfo<GatherTransformationParams> obj) {
+        const ngraph::PartialShape inputShape = std::get<0>(obj.param);
+        const GatherTransformationTestValues testValues = std::get<1>(obj.param);
+
+        std::ostringstream result;
+        result << "_" << 
+            inputShape << "_" <<
+            testValues.gatherIndicesValues << "_" <<
+            testValues.actual.precisionBeforeDequantization << "_" <<
+            testValues.actual.dequantization << "_" <<
+            testValues.expected.dequantizationBefore;
+        return result.str();
+    }
+};
+
+TEST_P(GatherTransformation, CompareFunctions) {
+    InitNodeInfo().run_on_model(actualFunction);
+    actualFunction->validate_nodes_and_infer_types();
+    auto res = compare_functions(actualFunction, referenceFunction, true, true);
+    ASSERT_TRUE(res.first) << res.second;
+
+    ASSERT_TRUE(LayerTransformation::allNamesAreUnique(actualFunction)) << "Not all names are unique";
+}
+
+namespace testValues1 {
+const std::vector<ngraph::PartialShape> inputShapes3D = {
+    { 3, 3, 4 }
```

Comment:
Could you please add dynamic shape instances?

Reply:
Done. Thanks!